src/supervisor/metadata-extractor.ts

```diff
@@ -143,6 +149,14 @@ export async function buildMetadataForWorkload(
   const isAssociatedWithParent = isPodAssociatedWithParent(pod);
 
   if (!pod.metadata || pod.metadata.namespace === undefined || !pod.spec) {
+    const logContext = {
+      podMetadata: pod.metadata,
+      podSpec: pod.spec,
```

Comment:
Both of these can hold a lot of data. Instead could we log only selected fields that can help us with debugging? For example, the workload name, type and namespace, and its images